pkg/virtual/syncer/controllers/apireconciler/syncer_apireconciler_reconcile.go

```diff
@@ -28,116 +28,95 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/tools/clusters"
 	"k8s.io/klog/v2"
 	"k8s.io/kube-openapi/pkg/common"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	generatedopenapi "k8s.io/kubernetes/pkg/generated/openapi"
 
 	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
-	"github.com/kcp-dev/kcp/pkg/reconciler/workload/apiexport"
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/logging"
 	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apidefinition"
 	dynamiccontext "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/context"
 	"github.com/kcp-dev/kcp/pkg/virtual/framework/internalapis"
 )
 
-func (c *APIReconciler) reconcile(ctx context.Context, apiExport *apisv1alpha1.APIExport, apiDomainKey dynamiccontext.APIDomainKey, syncTargetWorkspace logicalcluster.Name, syncTargetName string) error {
-	if apiExport == nil || apiExport.Status.IdentityHash == "" {
-		// new APIExport that is not ready yet, or export got deleted
-
-		c.mutex.RLock()
-		_, found := c.apiSets[apiDomainKey]
-		c.mutex.RUnlock()
-
-		if !found {
-			klog.V(3).Infof("No APIs found for API domain key %s", apiDomainKey)
-			return nil
-		}
-
-		// remove the APIDomain
-		c.mutex.Lock()
-		defer c.mutex.Unlock()
-		klog.V(2).Infof("Deleting APIs for API domain key %s", apiDomainKey)
-		delete(c.apiSets, apiDomainKey)
-		return nil
-	}
-
-	if apiExport.ObjectMeta.Name != apiexport.TemporaryComputeServiceExportName {
-		// this is not something we're handling in this controller
+func (c *APIReconciler) reconcile(ctx context.Context, apiDomainKey dynamiccontext.APIDomainKey, syncTarget *workloadv1alpha1.SyncTarget) error {
+	if len(syncTarget.Status.SyncedResources) == 0 {
```

Comment:
Do we really want to do that here ?
What about the built-in resources (like coinfig maps, secrets, ...)
It seems to me that an APIefinitionSet with only built-in resources might make sense (even though not very useful ;-) )

Reply:
right, removed this part.